src/Castle.Windsor/MicroKernel/Arguments.cs

```diff
@@ -26,12 +26,8 @@ namespace Castle.MicroKernel
 	/// </summary>
 	public class Arguments : IDictionary
 	{
-		protected IDictionary arguments;
-
-		public Arguments(object namedArgumentsAsAnonymousType, params IArgumentsComparer[] customComparers)
-			: this(new ReflectionBasedDictionaryAdapter(namedArgumentsAsAnonymousType), customComparers)
-		{
-		}
+		private bool isReadOnly;
+		private readonly IDictionary arguments;
 
 		public Arguments(IDictionary values, params IArgumentsComparer[] customComparers)
```

Comment:
I had 3 goes at trying to make this work. 

The problem is to do with the generic dictionary definition itself. It is `new Dictionary<object, object>`. To make it work with `StringComparison` that would involve instantiating it as `new Dictionary<string, object>`. 

It turns into a blood bath when running the unit tests. 

An example of where you can see how the dictionary needs to have a key of an `object` type is in [DefaultComponentActivator](https://github.com/castleproject/Windsor/blob/master/src/Castle.Windsor/MicroKernel/ComponentActivator/DefaultComponentActivator.cs#L62). Here the instance itself is getting used as the key. You can see the implementation of [SetContextualProperty](https://github.com/castleproject/Windsor/blob/master/src/Castle.Windsor/MicroKernel/Context/CreationContext.cs#L284) here. 

This would be awesome to sort out. It is unfortunately beyond the scope of this PR.

Reply:
Yeah, almost looked like it was possible. So I appreciate you asking. Always to try and get rid of things we don't need.

Reply:
> Yeah, almost looked like it was possible. So I appreciate you asking. Always good to try and get rid of things we don't need.

Did I miss something? I still don't see how they are used (other than a few pointless unit tests making all strings equal).

The only `Arguments` constructor usage I see useful at the moment is by providing no `IArgumentsComparer`s, which results in the internal dictionary of `Arguments` being initialised with an instance of `ArgumentsComparer` which is a `System.Collections.Generic.IEqualityComparer<object>` not an `IArgumentsComparer`, this comparer is the one that handles case-insensitive string keys and does normal `object.Equals` for types.

Reply:
I am also going to update the CHANGELOG.md with the breaking change.

Reply:
Great 👍